File: 050.Pow-x-n/wyrj/Pow-x-n.js
/**
 * @param {number} x
 * @param {number} n
 * @return {number}
 */
var myPow = function(x, n) {
    if (n < 0) {
        return 1 / myPow(x, -n);
    }
    if (0 === n) {
        return 1;
    }
    var sq;
    if (0 === n % 2) {
        return myPow(x * x, Math.floor(n / 2));
    } else {
        return myPow(x * x, Math.floor(n / 2)) * x;
    }
};

File: 050.Pow-x-n/Ddfish/Pow-x-n.swift
class Solution {
    func myPow(_ x: Double, _ n: Int) -> Double {
    
        if( x == 0 ) { return 0 }
        if( n == 0 ) { return 1 }
        if( n < 0 )  { return  1 / myPow(x,n*(-1))}
        let y = myPow(x, n/2)
        if n % 2 == 0 {
            return y * y
        }else {
            return x * y * y
        }


    }
}
